mlir/lib/Dialect/Vector/Transforms/VectorTransforms.cpp

```diff
@@ -2056,8 +2056,11 @@ void mlir::vector::populateShapeCastFoldingPatterns(RewritePatternSet &patterns,
 
 void mlir::vector::populateDropUnitDimWithShapeCastPatterns(
     RewritePatternSet &patterns, PatternBenefit benefit) {
-  patterns.add<DropUnitDimFromElementwiseOps, DropUnitDimsFromTransposeOp,
-               ShapeCastOpFolder, DropUnitDimsFromScfForOp>(
+  // TODO: Consider either including DropInnerMostUnitDimsTransferRead,
+  // DropInnerMostUnitDimsTransferWrite or better naming to distinguish this
+  // and populateVectorTransferCollapseInnerMostContiguousDimsPatterns.
```

Comment:
Should the comment belong to `transform::ApplyDropUnitDimWithShapeCastPatternsOp::populatePatterns`?

Reply:
It's hard to tell - _this file_ is where e.g. `DropInnerMostUnitDimsTransferWrite` and `DropUnitDimFromElementwiseOps` are actually added 😅 But then this is just the implementation and the interfaces is elsewhere.

Let me keep it here - I intend to review the corresponding tests in the coming days and will get rid of this.  Also, changes in this file are NFCs and not required (added this comment + ordered the patterns alphabetically). I'll add a note in the summary.